asyncapi/wso2.identity.server/dispatcher_service.bal

```diff
@@ -0,0 +1,116 @@
+        byte [] output = check crypto:hmacSha256(binaryPay, self.listenerConfig.webhookSecret.toBytes());
+        string computedDigest = output.toBase16();
+        if (trimmedSecret.length() != computedDigest.length() || trimmedSecret !== computedDigest) {
+            // Validate secret with x-wso2-event-signature header for intent verification.
+            log:printError("Signature verification failure");
+            http:Response response = new;
+            response.statusCode = http:STATUS_UNAUTHORIZED;
+            response.setPayload("Signature verification failure");
+            return response;
+        }
+
+        GenericPayloadType|error genericPayloadType = payload.cloneWithType(GenericPayloadType);
+        if (genericPayloadType is error) {
+            log:printError("Unsupported payload");
+            return genericPayloadType;
+        } else {
+            check self.matchRemoteFunc(genericPayloadType);
+            http:Response response = new;
+            response.statusCode = http:STATUS_OK;
+            response.setPayload("Event acknoledged successfully");
+            return response;
+        }
+    }
+
+    private function matchRemoteFunc(GenericPayloadType genericPayloadType) returns error? {
+        json events = genericPayloadType.events;
+        map<json> eventMap = <map<json>>events;
+        string eventKey = eventMap.keys()[0];
+        string[] mainEventParts = regex:split(eventKey, "/events/");
+        string mainEvent = regex:split(mainEventParts[1], "/event-type/")[0];
+        string[] eventTypeParts = regex:split(eventKey, "/event-type/");
+        string eventType = eventTypeParts[1];
+        log:printInfo("Received event: " + mainEvent + " of type: " + eventType);
+        match mainEvent {
+            "user" => {
+                match eventType {
+                    "userCreated" => {
+                        json eventData = eventMap[eventKey];
+                        GenericUserOperationData userCreatedData = check eventData.cloneWithType(GenericUserOperationData);
+                        GenericSecurityData securityData = {
+                            iss: genericPayloadType.iss,
+                            jti: genericPayloadType.jti,
+                            iat: genericPayloadType.iat,
+                            rci: genericPayloadType.rci
+                        };
+                        GenericUserOperationEvent userCreatedEvent = {
+                            eventData: userCreatedData,
+                            securityData: securityData
+                        };
+                        check self.executeRemoteFunc(userCreatedEvent, "userCreated", "UserOperationService", "onCreateUser");
+                    }
+                }
+            }
+        }
+    }
+
+    private function executeRemoteFunc(GenericDataType genericEvent, string eventName, string serviceTypeStr, string eventFunction) returns error? {
```

Comment:
this line exceeds 120 characters

Reply:
Fixed with https://github.com/ballerina-platform/asyncapi-triggers/pull/171/commits/8c04fdcc09d1d8ea40eeae48e2ce62784d61a66c